src/common/raft/integration/cagra_index.cu

```diff
@@ -14,8 +14,17 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-#include "common/raft/proto/raft_index_kind.hpp"
+#undef RAFT_EXPLICIT_INSTANTIATE_ONLY
 #include "common/raft/integration/raft_knowhere_index.cuh"
+#include "common/raft/proto/filtered_search_instantiation.cuh"
+#include "common/raft/proto/raft_index_kind.hpp"
+
+RAFT_FILTERED_SEARCH_EXTERN(cagra, raft_knowhere::raft_data_t<raft_proto::raft_index_kind::cagra>,
```

Comment:
Unfortunately due to an unresolved issue in RAFT, filtered searches trigger an extensive rebuild of RAFT components. Here, we declare extern templates for all of the instantiations we wish to use, which allows us to avoid triggering rebuilds unnecessarily.